core/controllers/admin.py

```diff
@@ -335,3 +336,50 @@ def get(self):
             'attachment; filename=topic_similarities.csv')
         self.response.write(
             recommendations_services.get_topic_similarities_as_csv())
+
+
+class DataExtractionQueryHandler(base.BaseHandler):
+
+    @require_super_admin
+    def post(self):
+        exp_id = self.payload.get('exp_id')
+        exp_version = self.payload.get('exp_version')
+        state_name = self.payload.get('state_name')
+        num_answers = self.payload.get('num_answers')
+
+        exploration = exp_services.get_exploration_by_id(
+            exp_id, strict=False, version=exp_version)
+
+        if exploration is None:
+            response = {
+                'success': False,
+                'message': 'no exploration with ID \'%s\' exist.' % exp_id,
+                'data': None
+            }
+            self.render_json(response)
+            return
+
+        if state_name not in exploration.states.keys():
+            response = {
+                'success': False,
+                'message': (
+                    'exploration \'%s\' does not have \'%s\' state.'
+                    % (exp_id, state_name)),
+                'data': None
+            }
+            self.render_json(response)
+            return
+
+        state_answers = stats_services.get_state_answers(
+            exp_id, exp_version, state_name)
+        extracted_answers = state_answers.get_submitted_answer_dict_list()
+
+        if num_answers > 0:
+            extracted_answers = extracted_answers[:num_answers]
```

Comment:
No. `stats_services.get_state_answers` returns all the answers associated with a state (so includes all shards). There is no method (which is public) implemented to get a specific shard of `StateAnswerModel`. I think this was adopted because they did not want to expose the structure of answers are stored to outside of the model. So you can either get master model (which contaisn shard_count) or you have to get all shards. Therefore I am using the `stats_services.get_state_answers` function which unifies all the answers stored in all shards in a domain object.